src/Gizmos/planeRotationGizmo.ts

```diff
@@ -201,6 +269,100 @@ export class PlaneRotationGizmo extends Gizmo {
+
+        // Normalize v2
+        const v2mag = Math.sqrt(v2.x * v2.x + v2.y * v2.y + v2.z * v2.z);
+        const v2norm = new Vector3(v2.x / v2mag, v2.y / v2mag, v2.z / v2mag);
+
+        // Calculate the dot products of vectors v1 and v2
+        const dotProducts = v1norm.x * v2norm.x + v1norm.y * v2norm.y + v1norm.z * v2norm.z;
+        const cross = Vector3.Cross(v1norm as any, v2norm as any);
+
+        // Extract the angle from the dot products
+        let angle = (Math.acos(dotProducts) * 180.0) / Math.PI;
+        angle = Math.round(angle * 1000) / 1000;
+        angle = Angle.FromDegrees(angle).radians();
+
+        // Flip if its cross has negitive y orientation
+        if (cross.y < 0) { angle = -angle; }
+
+        return angle;
+    }
+
+    private setupRotationCircle(paths: any[], parentMesh: AbstractMesh): Mesh {
+        const fillRadians = 0;
+        const step = this.circleConstants.pi2 / this.circleConstants.tessellation;
+        for (let p = -Math.PI / 2; p < Math.PI / 2 - 1.5; p += step / 2) {
+            const path = [];
+            for (let i = 0; i < this.circleConstants.pi2; i += step) {
+                if (i < fillRadians) {
+                    const x = this.circleConstants.radius * Math.sin(i) * Math.cos(p);
+                    const z = this.circleConstants.radius * Math.cos(i) * Math.cos(p);
+                    const y = 0;
+                    path.push(new Vector3(x, y, z));
+                } else {
+                    path.push(new Vector3(0, 0, 0));
+                }
+            }
+
+            paths.push(path);
+        }
+
+        const mat = new StandardMaterial("", this.gizmoLayer.utilityLayerScene);
+        mat.diffuseColor = Color3.Yellow();
+        mat.backFaceCulling = false;
+        const mesh = Mesh.CreateRibbon("ignore", paths, false, false, 0, this.gizmoLayer.utilityLayerScene, true);
+        mesh.material = mat;
+        mesh.material.alpha = .25;
+        mesh.rotation.x = Math.PI / 2;
+        parentMesh.addChild(mesh);
+        return mesh;
+    }
+
+    private updateRotationPath(pathArr: any[], newFill: number): void {
+        // To update the Ribbon, you have to mutate the pathArray in-place
+        const step = this.circleConstants.pi2 / this.circleConstants.tessellation;
+        let tessellationCounter = 0;
+        for (let p = -Math.PI / 2; p < Math.PI / 2 - 1.5; p += step / 2) {
+            const path = pathArr[tessellationCounter];
+            if (path) {
+                let radianCounter = 0;
+                for (let i = 0; i < this.circleConstants.pi2; i += step) {
+                    if (path[radianCounter]) {
+                        if (i < Math.abs(newFill)) {
+                            const eie = (newFill > 0) ? i : i * -1;
+                            const pea = (newFill > 0) ? p : p * -1;
+                            path[radianCounter].x = this.circleConstants.radius * Math.sin(eie) * Math.cos(pea);
```

Comment:
using the vector's `set` function will be a bit quicker, as `x`, `y`, and `z` are actually getters/setters and not class members